src/testing/resolvers/pools.ts

```diff
@@ -46,6 +53,19 @@ const poolsResolvers = {
         return HttpResponse.json(error, { status: error.code });
       }),
   },
+  deletePool: {
+    resolved: false,
+    handler: () =>
+      http.delete(`${BASE_URL}MAAS/a/v3/resource_pools/:id`, () => {
+        poolsResolvers.deletePool.resolved = true;
+        return HttpResponse.json({}, { status: 204 });
+      }),
+    error: (error: DeleteResourcePoolError = mockDeletePoolError) =>
+      http.delete(`${BASE_URL}MAAS/a/v3/resource_pools/:id`, () => {
+        poolsResolvers.deletePool.resolved = true;
+        return HttpResponse.json(error, { status: 404 });
```

Comment:
I would make the code overridable here as well. Not essential but just nice-to-have (:
```suggestion
        return HttpResponse.json(error, { status: error.code });
```